.github/actions/dockerfiles/Dockerfile.alpine-binary

```diff
@@ -10,13 +10,12 @@ ARG TARGETVARIANT
 ARG REPO=stacks-network/stacks-core
 
 RUN case ${TARGETARCH} in \
-    "amd64")  BIN_ARCH=linux-musl-x64  ;; \
+    "amd64")  BIN_ARCH=linux-musl-x64-v3  ;; \
```

Comment:
there will still be a legacy *build*, the change here is specific to the release docker image. 
i.e. all future docker images built for x64 will be the binaries built with x86-64-v3 build arg. 

https://github.com/docker-library/official-images?tab=readme-ov-file#architectures-other-than-amd64

if you visit the link to the sample run i did on my fork simulating a release, you'll see there are 2 x64 binary archives, 1 for legacy, and 1 for v3. 
i'm proposing with this change that for release docker images, we default to x86-64-v3 binaries. 
<img width="1083" alt="Screenshot 2024-03-01 at 17 08 27" src="https://github.com/stacks-network/stacks-core/assets/2847772/078f5518-8c89-4678-a03b-9044a4a448ab">

Reply:
i'll have to dig into this, but it may be possible to also build the docker image for x86-64 legacy. 
the question is if it's worth the time it will take - i'll do some tests locally to see what's involved here, but i have  feeling the amount of users looking for that legacy x64 image would be quite small

Reply:
IMO not worth it. I really doubt the people running the Docker builds are doing it on ancient hardware

Reply:
Seems like `v3` is supported in the VMs Hiro's cloud infra uses, but cannot speak for others:


```
/lib/x86_64-linux-gnu/ld-linux-x86-64.so.2 --help
...
  x86-64-v4
  x86-64-v3 (supported, searched)
  x86-64-v2 (supported, searched)
```

Reply:
added enough logic in latest commit so we can re-add v2 docker images/binaries if needed